### PR TITLE
Prefix all routes via path_prefix setting

### LINKED ISSUE #513: Support non-root API route
#### Problem description

👋  Hello all!

I'd like to deploy TiTiler to a non-root (non-`/`) AWS API Gateway route as a proxy Lambda. This currently results in `Not Found` responses, presumably because there's no TiTiler configuration to support a root-level FastAPI prefix. For example, on a `/titiler` route:

```
$ curl 'https://api.danschoppe.com/titiler/healthz'
{"detail":"Not Found"}
```

I see TiTiler is already making good use of [FastAPI's prefix](https://fastapi.tiangolo.com/tutorial/bigger-applications/#another-module-with-apirouter) for intermediate routers in the app, like `/cog/*` and `/stac/*`.

**Would the project be open to a `path_prefix` property in `ApiSettings` that would be applied to all TiTiler routes?** Users could configure the prefix with a `TITILER_API_PATH_PREFIX` environment variable.

~I've [implemented the feature on a fork](https://github.com/SenteraLLC/titiler/compare/master...SenteraLLC:titiler:path-prefix) (please disregard the bits of the diff that have to do with building a Lambda `.zip` file) to test such a feature and illustrate how a PR might look.~
I've opened #515 to demonstrate how this looks.

Thanks for everything you do!

#### Expected Output

```
$ curl 'https://api.danschoppe.com/titiler/healthz'
{"ping":"pong!"}
```

EDIT: I didn't know if a GitHub issue was the appropriate place for this. Let me know if you'd like me to move it to a discussion topic.

### PR DESCRIPTION
**Pull request by DanSchoppe**
## What I am changing
This PR introduces the ability to host TiTiler on a non-root (non-`/`) API route.

## How I did it
1. Add support for a `path_prefix` setting in `src/titiler/application/titiler/application/settings.py`. This can be set via the `TITILER_API_PATH_PREFIX` environment variable.
2. Make use of the `ApiSettings().path_prefix` in `src/titiler/application/titiler/application/main.py` by concatenating it with the existing prefixes (eg. `"cog"` and `"stac"`).

## How you can test it
Deploy TiTiler as a proxy Lambda to a non-root AWS API Gateway route. Query the various endpoints to confirm they resolve correctly.

## Related Issues
Closes #513

### CODE REVIEW

**Comment by vincentsarago** [comment]
do you think we can test this locally?

**Comment by DanSchoppe** [comment]
> do you think we can test this locally?

I would imagine so — I'll get started on that. Assuming the tests go well, do you imagine this will be a welcome enhancement?

Other than local test(s), do you imagine any other required changes? Documentation? AWS CDK? I don't use the CDK so that would be slightly more challenging.

**Comment by vincentsarago** [comment]
just python test would be fine

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ repos:
         language_version: python
 
   - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.812
+    rev: v0.971
```
Comment (DanSchoppe) [inline .pre-commit-config.yaml:27]: This fixes a compilation error when `pre-commit` attempts to install `mypy` on my M2 MacBook Pro. The `pre-commit` author on Stack Overflow [suggests](https://stackoverflow.com/a/71298400) fixing the issue by updating to a recent version of mypy.

File: src/titiler/application/titiler/application/settings.py [hunk 2]
```diff
@@ -10,6 +10,7 @@ class ApiSettings(pydantic.BaseSettings):
     cors_origins: str = "*"
     cachecontrol: str = "public, max-age=3600"
     root_path: str = ""
+    path_prefix: str = ""
```
Comment (DanSchoppe) [inline src/titiler/application/titiler/application/settings.py:13]: @vincentsarago I believe I've covered all configuration changes and tests for API routes:
- `/cog`
- `/stac`
- `/mosaicjson`
- tms
- `/`
- `/healthz`

I *haven't* accounted for the `/docs` endpoint (or docs links). Should I dig into that?

**Comment by vincentsarago** [comment]
@DanSchoppe I'm rushing on other project for now, I'll keep this on top of my TODO list. I hope it's not blocking you

**Comment by DanSchoppe** [comment]
> @DanSchoppe I'm rushing on other project for now, I'll keep this on top of my TODO list. I hope it's not blocking you

Yep, no problem.

**Comment by vincentsarago** [comment]
@DanSchoppe I'm really sorry I'm trying to finish rio-tiler big update before I want to do any change in titiler at this point.

**Comment by DanSchoppe** [comment]
> @DanSchoppe I'm really sorry I'm trying to finish rio-tiler big update before I want to do any change in titiler at this point.

Yep, no hurry and no need to apologize! Didn't know if this fell off your radar so I was just checking in. Good luck with your rio-tiler push!

**Comment by vincentsarago** [comment]
@DanSchoppe looking at this again and I think this should be fixed at the Lambda handler level by using `api_gateway_base_path=` in Mangum and the setting `TITILER_API_ROOT_PATH=`

see: https://github.com/jordaneremieff/mangum/issues/142

**Comment by DanSchoppe** [comment]
Thanks @vincentsarago ! Mind if this stays open for a few days while I wrap something up, then I'll circle back on the Mangum approach?

**Comment by vincentsarago** [comment]
@DanSchoppe no worry!

**Comment by DanSchoppe** [comment]
Blah, I'm not going to get to this anytime soon. I'll go ahead and close this and open something fresh down the road.

**Pull request closed without merging**